### DISCUSSION TITLE
Best way to reboot a caprover server?

### DISCUSSION THREAD

**Discussion by tinmrn** [caprover/caprover#1554] [Q&A] [answered]
One of my containers is MongoDB and I would like it to shutdown gracefully before I reboot the server.

For other apps it doesn't matter because they are (and should be) stateless, but for some containers that have persistent storage it's just nicer to give them a regular stop command.

Now I do this before shutdown/reboot:

`docker service ls -q|xargs -I {} docker service scale -d {}=0`

And then this after the server comes back:

`docker service ls -q|xargs -I {} docker service scale -d {}=1`

Is there a better way?

**Comment by githubsaturn** [comment, marked as answer]
Technically docker should forward the signals to your containers and they should be shutting down gracefully. But yea your approach works as a safe backup.

**Reply by tinmrn** [reply]
Cool thanks.

I've also increased the wait period on my mongo container with

`docker service update --stop-grace-period 5m srv-captain--mongo-01`

**Reply by dawidmachon** [reply]
@githubsaturn yeah, right now mongo get gracefull exit sign, but I got one issue with it in past - it was too fast ;)
